src/vm/excep.cpp

```diff
@@ -7280,8 +7280,17 @@ bool IsIPInMarkedJitHelper(UINT_PTR uControlPc)
 {
     LIMITED_METHOD_CONTRACT;
 
+#ifdef _ARM_
+    // The least bit denoting THUMB might confuse the range check because
+    // PC values have the least bit reset. It causes a bug if PC is at
+    // the first byte of the function range.
+#define CHECK_RANGE(name) \
+    if ((GetEEFuncEntryPoint(name)|0x1) <= (uControlPc|0x1) && (uControlPc|0x1) < (GetEEFuncEntryPoint(name##_End)|0x1)) return true;
```

Comment:
`GetEEFuncEntryPoint` returns `PCODE`. `PCODE` is expected to have the thumb bit set (https://github.com/dotnet/coreclr/blob/master/src/inc/daccess.h#L2410).

There is `PCODEToPINSTR` macro that should be used to do the kind of adjustment like what you are doing here, without ifdefs.

However, it may be better to set the thumb bit during unix context -> CONTEXT conversion in PAL instead. `CONTEXT::Pc` has the bit set on Windows ARM, and it is likely that number of random places like this one may took subtle decency on it. Setting the bit in the PAL will save you from chasing all of them down.